### DISCUSSION TITLE
java.lang.IllegalStateException: Registry is already frozen

### DISCUSSION THREAD

**Discussion by AnAwesomGuy** [FabricMC/fabric-api#3187] [Mod Dev Support] [answered]
### Description

[crash report](https://github.com/FabricMC/fabric/files/12023923/crash-2023-07-12_13.04.35-client.txt) and [class](https://github.com/FabricMC/fabric/files/12023932/StoneShears.txt) attached
this started happening when i removed fabric api bcuz i didnt need it
asking here instead of discord cuz i cant use discord in china

**Comment by Jab125** [comment, marked as answer]
You need Fabric API which fixes the issue
(https://github.com/FabricMC/fabric/tree/1.19.4/fabric-registry-sync-v0 contains the fix)

**Reply by Jab125** [reply]
alternatively you could depend and include on the registry sync module (which depends on fabric-api-base and fabric-networking-api-v1 as shown [here](https://github.com/FabricMC/fabric/blob/1.19.4/fabric-registry-sync-v0/build.gradle).
gradle dependencies:
```gradle
dependencies {
    modImplementation include(fabricApi.module("fabric-registry-sync-v0", project.fabric_version))
    include fabricApi.module("fabric-api-base", project.fabric_version) // just to include it into the jar
    include fabricApi.module("fabric-networking-api-v1", project.fabric_version) // just to include it into the jar
}
```

**Reply by AnAwesomGuy** [reply]
ill just make it require fabric api ig, most people already have fabric api anyway
